### PR TITLE
refactor(android): improve parity for keyboardType

### PR DESCRIPTION
**Pull request by ypbnv**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-17203

**Description:**
Refactor kyeboardType handling in order to get rid off redundant code and improve parity. Assign the TYPE_CLASS_TEXT only for text type input types.
Note: No unit tests since this is visual change.

@vijaysingh-axway Would you be able to check two things on the iOS related to the ticket - is the TYPE_NAMEPHONE_PAD constant used, because I see it is not taken into account in the Android code? Also is the expected behavior in iOS to be able to put multiple decimal points for TYPE_DECIMAL_PAD? That would not be a native android behavior. Setting the decimal tab flag only allows one decimal point.

**Test case:**
_app.js_
```js
var win = Ti.UI.createWindow({layout:'vertical'}),
	textASCII = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_ASCII}),
	textDecimalPad = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_DECIMAL_PAD}),
	textDefault = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_DEFAULT}),
	textEmail = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_EMAIL}),
	textNamephonePad = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_NAMEPHONE_PAD}),
	textNumbersPunctuation = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_NUMBERS_PUNCTUATION}),
	textNumberPad = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_NUMBER_PAD}),
	textPhonePad = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_PHONE_PAD}),
	textURL = Ti.UI.createTextField({width: 300, keyboardType: Titanium.UI.KEYBOARD_TYPE_URL});
win.add([textASCII,
		textDecimalPad,
		textDefault,
		textEmail,
		textNamephonePad,
		textNumbersPunctuation,
		textNumberPad,
		textPhonePad,
		textURL]);
win.open();

```

TYPE_NUMBER_PAD and TYPE_DECIMAL_PAD should no more allow typing non-number characters, besides one decimal point for the latter.

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java [hunk 1]
```diff
@@ -71,17 +71,6 @@
...
-	private static final int KEYBOARD_NUMBER_PAD = 3;
-	private static final int KEYBOARD_PHONE_PAD = 4;
-	private static final int KEYBOARD_EMAIL_ADDRESS = 5;
-	@SuppressWarnings("unused")
-	private static final int KEYBOARD_NAMEPHONE_PAD = 6;
-	private static final int KEYBOARD_DEFAULT = 7;
-	private static final int KEYBOARD_DECIMAL_PAD = 8;
-
 	// UIModule also has these as values - there's a chance they won't stay in sync if somebody changes one without changing these
 	private static final int TEXT_AUTOCAPITALIZATION_NONE = 0;
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java]: We should probably remove these constants too, and use `UIModule`.

**Review by garymathews — CHANGES_REQUESTED** [review]

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java (hunk 1 above)
Comment (garymathews) [reply android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java]: Same for `RETURNKEY_*` constants

Comment (ypbnv) [reply android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java]: Removed both blocks and pointed the constant to the UIModule namespace.

**Comment by sgtcoolguy** [comment]
ping @ypbnv  and @garymathews - have the changes satisfied the CR? Can we dismiss it and move to in qe testing?

**Review by garymathews — APPROVED** [review]
CR: **PASS**

**Comment by vijaysingh-axway** [comment]
Answer for iOS related query-
1. KEYBOARD_TYPE_NAMEPHONE_PAD is used in iOS.
2. KEYBOARD_TYPE_DECIMAL_PAD - It is possible to put multiple decimal point.

**Comment by ssekhri** [comment]
FR Passed. The KEYBOARD_TYPE_NUMBER_PAD does not accept decimal point. The KEYBOARD_TYPE_DECIMAL_PAD is still different on iOS and Android as it accepts multiple points in iOS but just one in Android but these are by native behaviors.
Verified on:
Mac OS: 10.15.1
SDK: 9.1.0.v20200325104908
Appc CLI: 8.0.0
JDK: 11.0.4
Node: 10.17.0
Studio: 6.0.0.202003181504
Device: Nexus 4(v5.1)

**Pull request merged**